src/java/org/apache/cassandra/index/sai/utils/PriorityQueueIterator.java

```diff
@@ -18,24 +18,30 @@
 
 package org.apache.cassandra.index.sai.utils;
 
+import java.util.Iterator;
 import java.util.PriorityQueue;
 
 /**
  * An iterator that returns elements from a priority queue in order.
  */
-public class PriorityQueueOrderIterator extends OrderIterator
+public class PriorityQueueIterator<T> implements Iterator<T>
```

Comment:
sweet

Reply:
maybe worth making explicit -- "unlike pq.iterator(), this will destructively poll the pq for its contents in priority order"